package.json

```diff
@@ -9,7 +9,7 @@
     "build-spec": "YEAR=2020 && git checkout --quiet \"es${YEAR}\" && mkdir -p \"out/${YEAR}\" && cp -R img \"out/${YEAR}\" && ecmarkup --verbose spec.html \"out/${YEAR}/index.html\" --css \"out/${YEAR}/ecmarkup.css\" --js \"out/${YEAR}/ecmarkup.js\" && git checkout --quiet travis-origin/test-travis",
     "postbuild-spec": "git remote rm travis-origin",
     "prebuild-master": "npm run clean && mkdir out && cp -R img out",
-    "build-master": "npm run wipe-es6biblio && ecmarkup --verbose spec.html out/index.html --css out/ecmarkup.css --js out/ecmarkup.js",
+    "build-master": "npm run wipe-es6biblio && ecmarkup --lint-spec --verbose spec.html out/index.html --css out/ecmarkup.css --js out/ecmarkup.js",
```

Comment:
rather than add this in `build-master`, perhaps this should be added as a travis job, so it can run in parallel with the build process?

Reply:
The linting isn't really divorced from the building, since they do much of the same processing. And if you pass `--lint-spec` and there are no linting errors it just builds. (Think of it like `-Wall -Werror`, if you're familiar with C++ compilers.) I don't think there's much to be gained from splitting it out as a separate job.

Reply:
i was thinking clearer error messages - ie, the lint failed but the build passed tells you more than "the build failed"

Reply:
@bakkot Thoughts on having a `--lint-only` that exits after linting, regardless of the number of errors?

Reply:
@michaelficarra Some of the [things I want to enforce](https://github.com/tc39/ecmarkup/issues/173) (e.g. the output is valid HTML) could not feasibly done without running the full build, so I'm not sure there would be much point.

Reply:
@bakkot That sounds like it could be the job of a separate, more generic tool.

Reply:
True.

I'm fine with adding `--lint-only`, I guess. I am still worried that we may want to add linting checks which are integrated with later steps of the compilation, which would make it less useful.

Reply:
@michaelficarra For a concrete example of a late-stage linting error, ecmarkup currently `console.log`s a warning when [it finds a duplicate id](https://github.com/tc39/ecmarkup/blob/c7e4349b54f7100009cdbdc5f81779e30be2b0e6/src/Builder.ts#L29) or [it can't link an xref](https://github.com/tc39/ecmarkup/blob/b27317bc9a78f67a04b2ace5babd0ba05e930f68/src/Xref.ts#L121). It seems like these should be linting errors instead. But building and xref linking happen relatively late in the compilation process.

Reply:
It'd still probably be useful to be able to do linting without having to output unwanted artifacts to the filesystem.